File: 백준/Silver/10828. 스택/스택.py
n = int(input())

process = []
for _ in range(n):
  process.append(list(input().split()))

stack = []
for p in process:
  if len(p) == 2:
    stack.append(p[1])
  else:
    if p[0] == 'pop':
      if len(stack) == 0:
        print(-1)
      else:
        print(stack.pop())
    elif p[0] == 'size':
      print(len(stack))
    elif p[0] == 'empty':
      if len(stack) == 0:
        print(1)
      else:
        print(0)
    else:
      if len(stack) == 0:
        print(-1)
      else:
        print(stack[-1])

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int n;
string command;
stack<int> st;

int main(void) {
	cin.tie(0)->sync_with_stdio(false);
	
	cin >> n;

	st = stack<int>();

	for (int i = 0; i < n; i++) {
		cin >> command;

		if (command == "push") {
			int s;
			cin >> s;
			st.push(s);
		}
		else if (command == "top") {
			if (!st.empty()) cout << st.top() << "\n";
			else cout << "-1" << "\n";
		}
		else if (command == "size") {
			cout << st.size() << "\n";
		}
		else if (command == "empty") {
			cout << st.empty() << "\n";
		}
		else {
			if (!st.empty()) {
				cout << st.top() << "\n";
				st.pop();
			}
			else {
				cout << "-1" << "\n";
			}
		}
	}

	return 0;
}